### PR TITLE
Adds a chance to fall while moving z-levels using zmove

### PR DESCRIPTION
**Pull request by Spookat**
Remember the central hole everyone treated as a jungle gym? This PR adds a small (but not inconsequential) chance to fail while instantly descending down lattices, catwalks and ladders. Climbing down ladders normally is still perfectly safe.

Failing in this manner will deal reduced damage in a fall (70% instead of 100%), and the chance is affected by certain species' climbing skills.

Now you can laugh at your colleagues for not following health & safety regulations!

### CODE REVIEW

File: code/modules/multiz/movement.dm [hunk 1]
```diff
@@ -89,6 +90,13 @@
 		return
 	. = ..()
 	if(.)
+		// Chance to fail while trying to climb down a lattice structure or ladder
+		var/turf/T = GetAbove(src)
+		if (locate(/obj/structure/lattice, T) || locate(/obj/structure/ladder, T) && !CanAvoidGravity() && !species.natural_climbing && (prob(8 * species.climb_coeff)))
```
Comment (mikomyazaki) [inline code/modules/multiz/movement.dm:95]: This probably also needs to check if there is gravity or not, otherwise they shouldn't be able to fall at all.

Comment (Spookat) [reply code/modules/multiz/movement.dm]: Good catch, added

**Comment by Spookat** [comment]
!review

File: code/modules/multiz/movement.dm [hunk 2]
```diff
@@ -89,6 +90,16 @@
 		return
 	. = ..()
 	if(.)
+		// Chance to fail while trying to climb down a lattice structure or ladder
+		var/turf/T = GetAbove(src)
+		var/area/A = get_area(src)
+		if (locate(/obj/structure/lattice, T) || locate(/obj/structure/ladder, T))
```
Comment (mikomyazaki) [inline code/modules/multiz/movement.dm:96]: If this is only meant to happen when you move downwards, you'll need a check for `direction == DOWN` aswell.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/modules/multiz/movement.dm (hunk 2 above)
Comment (Spookat) [reply code/modules/multiz/movement.dm:96]: Added check.

**Review by mikomyazaki — COMMENTED** [review]
Not totally sure about the probability or damage levels involved. I think the same effect could be achieved with a lower chance & lower damage. But I'll leave that for a maintainer to think about. Code-wise this is fine.

We had more discussion about the reason for this PR [here](https://discord.com/channels/157516682288562176/302201161044328450/957367849024901180)

**Review by Alberyk — APPROVED** [review]

**Comment by NonQueueingMatt** [comment]
Will be discussed between maintainers.

**Comment by NonQueueingMatt** [comment]
Apologies for the delay.

The main issue we saw here was that antagonists commonly do/can/should use the lattices to traverse zlevels quickly. This would take away from that, and be generally a hassle for them other than crew.

This is unfortunately the part where a skill system would be good.

**Comment by Spookat** [comment]
Closing as there's no easy workaround that I can think of. Might reopen later.

**Pull request closed without merging**